advanced/index.php

```diff
@@ -1,15 +1,31 @@
 <?php
 /* Detailed Pi-Hole Block Page: Show "Website Blocked" if user browses to site, but not to image/file requests based on the work of WaLLy3K for DietPi & Pi-Hole */
 
+function validIP($address){
+	if (preg_match('/[.:0]/', $address) && !preg_match('/[1-9a-f]/', $address)) {
+		// Test if address contains either `:` or `0` but not 1-9 or a-f
+		return false;
+	}
+	return !filter_var($address, FILTER_VALIDATE_IP) === false;
+}
+
 $uri = escapeshellcmd($_SERVER['REQUEST_URI']);
 $serverName = escapeshellcmd($_SERVER['SERVER_NAME']);
 
 // Retrieve server URI extension (EG: jpg, exe, php)
+ini_set('pcre.recursion_limit',100);
 $uriExt = pathinfo($uri, PATHINFO_EXTENSION);
 
 // Define which URL extensions get rendered as "Website Blocked"
 $webExt = array('asp', 'htm', 'html', 'php', 'rss', 'xml');
 
+// Immediately quit since we didn't block this page (the IP address or pi.hole is explicitly requested)
+if(validIP($serverName) || $serverName === "pi.hole")
```

Comment:
* This is similar but different from [$AUTHORIZED_HOSTNAMES](https://github.com/pi-hole/AdminLTE/blob/master/scripts/pi-hole/php/auth.php#L30).  
* Also related, I access pihole through a custom virtualhost through the VIRTUAL_HOST php environment variable that my docker auto injects to the php env for me.  Not sure how many people are using it outside docker but it is injected into [$AUTHORIZED_HOSTNAMES](https://github.com/pi-hole/AdminLTE/blob/master/scripts/pi-hole/php/auth.php#L39).
  * I was considering refactoring VIRTUAL_HOST to be in SetupVars instead of the PHP env for unification with that style of variable storage.  
* I also have a open issue in the docker repo to [support more than one virtual host](https://github.com/diginc/docker-pi-hole/issues/109)

So with all that in mind, perhaps VIRTUAL_HOST could 'go away' from PHP and docker/anyone with hostnames could just append my custom VIRTUAL_HOST(s) to AUTHORIZED_HOSTS which is source from SetupVars in all php locations that need to know what a 'pihole' address/hostname is?